plugins/GSdx/GSSettingsDlg.cpp

```diff
@@ -372,6 +383,47 @@ void GSSettingsDlg::UpdateControls()
 		bool ocl = renderer == GSRendererType::DX9_OpenCL || renderer == GSRendererType::DX1011_OpenCL || renderer == GSRendererType::OGL_OpenCL;
 		bool null = renderer == GSRendererType::Null;
 
+		int selected_CRC = SendMessage(GetDlgItem(m_hWnd, IDC_CRC_LEVEL), CB_GETCURSEL, 0, 0);
+
+		if (m_prev_renderer == GSRendererType::Undefined)
+		{
+			m_prev_renderer = static_cast<GSRendererType>(i);
+		}
+		else if (renderer == GSRendererType::OGL_HW && m_prev_renderer != GSRendererType::OGL_HW)
+		{
+			if (selected_CRC == 3)
+			{
+				if (!m_non_default_CRC)
+				{
+					SendMessage(GetDlgItem(m_hWnd, IDC_CRC_LEVEL), CB_SETCURSEL, 2, 0);
+				}
+				m_non_default_CRC = false;
+			}
+			else if (selected_CRC == 2)
+			{
+				m_non_default_CRC = true;
+			}
+
+			m_prev_renderer = static_cast<GSRendererType>(i);
+		}
+		else if ((renderer == GSRendererType::DX9_HW || renderer == GSRendererType::DX1011_HW) && m_prev_renderer != GSRendererType::DX9_HW && m_prev_renderer != GSRendererType::DX1011_HW)
+		{
+			if (selected_CRC == 2)
+			{
+				if (!m_non_default_CRC)
+				{
+					SendMessage(GetDlgItem(m_hWnd, IDC_CRC_LEVEL), CB_SETCURSEL, 3, 0);
+				}
+				m_non_default_CRC = false;
+			}
+			else if (selected_CRC == 3)
+			{
+				m_non_default_CRC = true;
+			}
+
+			m_prev_renderer = static_cast<GSRendererType>(i);
+		}
```

Comment:
Maybe it would be worth having a dedicated function for these CRC Hack level management stuffs? (@turtleli - any opinion?)

I prefer separating them into a separate function because the code looks rather complicated in comparison to the other simpler and straightforward codes in `UpdateControls()``.